typescript/sdk/src/token/schemas.ts

```diff
@@ -63,12 +48,36 @@ export const TokenConfigSchema = z.discriminatedUnion('type', [
   SyntheticConfigSchema,
 ]);
 
-export const TokenRouterConfigSchema = z.intersection(
-  TokenConfigSchema,
-  routerConfigSchema,
+export const TokenRouterConfigSchema = TokenConfigSchema.and(
+  GasRouterConfigSchema,
 );
 
-export const WarpRouteDeployConfigSchema = z.record(
-  z.string(),
-  TokenRouterConfigSchema,
-);
+export const WarpRouteDeployConfigSchema = z
+  .record(TokenRouterConfigSchema)
+  .refine((configMap) => {
+    const entries = Object.entries(configMap);
+    return (
+      entries.some(
```

Comment:
would this work if the configMap had a mix of valid and invalid configs?

Reply:
quick failing test for you. we're probably not expecting this combo in prod?

```typescript
it.only('should throw if one of the configs is invalid', async () => {
    const collateral = {
      arbitrum: {
        ...config.arbitrum,
        type: TokenType.collateral,
        token: SOME_ADDRESS,
      },
      op: {
        ...config.arbitrum,
        type: TokenType.synthetic,
        // symbol: 'symbol',
      },
    };
    expect(WarpRouteDeployConfigSchema.safeParse(collateral).success).to.be
      .false;
});

```

Reply:
this is a valid config
the token metadata for the synthetic will be derived from the collateral token